ibis/backends/duckdb/__init__.py

```diff
@@ -237,31 +241,36 @@ def configure_connection(dbapi_connection, connection_record):
 
         super().do_connect(engine)
 
+    @staticmethod
+    def _sa_load_extensions(dbapi_con, extensions):
+        query = """
+        SELECT extension_name, aliases
+        FROM duckdb_extensions()
+        WHERE installed AND loaded
+        """
```

Comment:
The previous query would ignore extensions that weren't known to duckdb (`duckdb_extensions()` returns known or installed extensions), meaning you couldn't use it to install any custom extensions. I couldn't figure out a way to do the filtering I wanted fully in SQL. The data we're querying is super small though, so doing it manually in python should be fine.

Reply:
There may be a way to incorporate an `allow_unsigned_extensions` argument that would allow you to load your own extensions or third-party extensions, overriding the check or something. That might could get messy.

Reply:
I fiddled around with the query, and might need to think through it, but the small dataset makes me think it is okay. 

```python
con.sql("""SELECT extension_name 
           FROM duckdb_extensions() 
           UNION SELECT UNNEST(aliases) 
           FROM duckdb_extensions();"""
```

I was looking to see if the goal was to unnest the aliases (with the `installed.update`) we might could use the `UNNEST` function in SQL to reduce a little bit of the Python. 

It might need some tweaking with the installed and loaded predicates if those need to be used as part of an inner/outer query.

Reply:
Thanks for the suggestion (`UNNEST` is a new one for me), with that prompting I was able to get the set of installed extension names fully in SQL. Probably doesn't matter either way for performance, but it does simplify the code a bit.